lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+        top_level.style = match top_level_style {
+            token::TextStyle::FormatLine => Some(token::TextStyle::UnclosedLine),
+            _                            => Some(top_level_style),
+        };
+        let text    = self.text_state.unsafe_end_literal();
+        let literal = text.into();
+        self.append_token(literal);
+        self.pop_state();
+        match line_ending {
+            token::LineEnding::None => self.on_eof(reader),
+            _                       => self.block_state.push_line_ending(line_ending),
+        }
+    }
+
+    /// Submit a missing closing quote in a text line literal.
+    fn submit_missing_quote<R:LazyReader>(&mut self, new_line:token::LineEnding, reader:&mut R) {
+        trace!(self.logger,"Text::submit_missing_quote");
+        let matched_bookmark = self.bookmarks.matched_bookmark;
+        self.bookmarks.rewind(matched_bookmark, reader);
+        let current   = self.text_state.unsafe_current_mut();
+        current.style = Some(token::TextStyle::UnclosedLine);
+        let text      = self.text_state.unsafe_end_literal();
+        let literal   = text.into();
+        self.append_token(literal);
+        self.pop_state();
+        match new_line {
+            token::LineEnding::None => self.on_eof(reader),
+            _                       => self.block_state.push_line_ending(new_line),
+        }
+    }
+
+    /// Triggered when encountering the opening of a text block in a nested context.
+    fn on_text_block_nested<R:LazyReader>(&mut self, new_line:token::LineEnding, reader:&mut R) {
+        trace!(self.logger,"Text::on_text_block_nested");
+        let current      = self.consume_current();
+        let offset       = self.offset.consume();
+        let text         = &current[0..3];
+        let unrecognized = Token::Unrecognized(text,offset);
+        self.append_token(unrecognized);
+        self.on_newline_in_interpolate(new_line,reader);
+    }
+
+    // === Lines ===
+
+    /// Triggered when beginning a format line literal.
+    fn on_begin_format_line<R:LazyReader>(&mut self, _reader:&mut R) {
+        trace!(self.logger,"Text::on_begin_format_line");
+        let interpolate = self.text_interpolate;
+        let literal     = self.text_state.begin_literal();
+        literal.style   = Some(token::TextStyle::FormatLine);
+        if self.is_inside_state(interpolate) {
+            let nested_format_line = self.text_format_line_nested;
+            self.push_state(nested_format_line);
+        } else {
+            let format_line_state = self.text_format_line;
+            self.push_state(format_line_state);
+        }
+    }
+
+    /// Submits a text line literal.
+    fn submit_text_line<R:LazyReader>(&mut self, end_to:group::Identifier, _reader:&mut R) {
+        trace!(self.logger,"Text::submit_text_line");
+        let text          = self.text_state.unsafe_end_literal();
+        let token         = text.into();
```

Comment:
alignment - please check alignment of arguments and `=` in other files as well. I'm not writing comments about it anymore.